src/middlewared/middlewared/plugins/enterprise/m_series_nvdimm_firmware_freebsd.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding=utf-8 -*-
+import logging
+import os
+import re
+
+import sysctl
+
+from middlewared.service import CallError, private, Service
+
+logger = logging.getLogger(__name__)
+
+
+class EnterpriseService(Service):
+
+    SIZE = None
+    SIZE_ERROR = "Size not retrieved yet"
+
+    VERSION = None
+    VERSION_ERROR = "Version not retrieved yet"
+
+    @private
+    def setup_m_series_nvdimm(self):
+        try:
+            self.SIZE = int(re.match("NVDIMM region ([0-9]+)GB", sysctl.filter("dev.nvdimm.0.%desc")[0].value).group(1))
+        except Exception as e:
+            self.SIZE_ERROR = str(e)
+
+        try:
+            # This should be ran before pool import so we cache the value obtained on first middlewared startup
+            cache_path = "/tmp/dev-nvdimm-0-dump_i2c"
+            if os.path.exists(cache_path):
+                with open(cache_path) as f:
+                    i2c_dump = f.read()
+            else:
+                i2c_dump = sysctl.filter("dev.nvdimm.0.dump_i2c")[0].value
```

Comment:
M60 is going to have two NVDIMMs per controller, so I guess we should check both?  Or we just assume that all new M60 are good enough and never going to have wrong version?

I also don't like the idea of reading full dev.nvdimm.0.dump_i2c, since it takes seconds for a run.  There is a method to read specific byte(s) I've pointed to Nick Wolff via setting dev.nvdimm.0.arg3 to ((offset << 8) | page) and then read the binary output from dev.nvdimm.0.func27, where first 4 bytes are status and following byte is the data.  I don't have NVDIMM inserted into my system right now to check, but it was working for me before, and that is what dump_i2c does inside for each byte.

Also I don't see any check for NVDIMM maker/model.  It makes me think it will fail if we switch to different one.  I am not sure right now what and how to check, or what models we are using, but IMO that definitely needs investigation.